core/frontend/src/types/notifications.ts

```diff
@@ -0,0 +1,77 @@
+import { Service } from './common'
+
+/** Levels of notifications available to be raised */
+export enum NotificationLevel {
+  Success = 'success',
+  Error = 'error',
+  Info = 'info',
+  Warning = 'warning',
+  Critical = 'critical',
+}
+
+
+/** Base Notification interface to be used for system-user communication */
+export interface NotificationInterface {
+  /**
+   * @param level - Indicates the nature of the notification. Similar to logging systems.
+   * @param service - The service that is pushing this notification
+   * @param type - Free string used for notification categorization
+   * @param message - What is being communicated
+   * @param time_created - UNIX timestamp indicating the time of creation
+  **/
+  level: NotificationLevel
+  service: Service
+  type: string
+  message: string
+  time_created: number
+}
+
+
+/** Base class for frontend notification system. */
+export class Notification implements NotificationInterface {
+  constructor (
+    public readonly level: NotificationLevel,
+    public readonly service: Service,
+    public readonly type: string,
+    public readonly message: string,
+    public readonly time_created: number,
+  ) {}
+
+  /** Returns a stringfied version of the object to be used as hash */
+  hash (): string {
+    return JSON.stringify(this)
+  }
+
+  /** Returns the time in seconds between time_created and given Date object */
+  secondsTill (date: Date): number {
+    const notification_time = new Date(this.time_created)
+    const time_diff = date.getTime() - notification_time.getTime()
+    return Math.floor(time_diff/1e3)
+  }
+
+  /** Compare if all properties (except time_created) between this and other match */
+  similarTo (other: Notification): boolean {
```

Comment:
`isEqual` maybe ? This make clear that will return a boolean

Reply:
Will change to `isSimilarTo`. Equal would not be a good fit here, since they are not exactly equal (their timestamps are different).